popup.js

```diff
@@ -0,0 +1,137 @@
+			initializeLogLink(date);
+			makeObjectInspectionCollapsible(date);
+			listeners.forEach(listenerEntry => {
+				document.getElementById(listenerEntry.elementId).addEventListener('click', listenerEntry.listener);
+			});
+		}
+	}
+);
+
+function formatServerResponse(serverResponse) {
+	let textParts = serverResponse.text.split('ManualSapTestInfo');
+	let text = textParts[0];
+	if (textParts.length > 1) {
+		text = textParts[0] +
+			'<div id="' + serverResponse.date + '-teaser">ManualSapTestInfo [<span style="color: darkgray">click to' +
+			' inspect</span>]</div>' +
+			'<div id="' + serverResponse.date + '-info" style="display: none">' +
+			'ManualSapTestInfo' + textParts[1].replace('[', '[<br>&nbsp;&nbsp;&nbsp;').replace(/, /g, '<br>&nbsp;&nbsp;&nbsp;') + '</div>';
+	}
+
+	text = text.replace(/(?:\r\n|\r|\n)/g, '<br>');
+
+	const resetUserStateUrl = 'api/projects/{project}/sap-test-event/reset/{sap-user}';
+	text = text.replace(resetUserStateUrl, resetUserStateUrl + ' (<span class="reset-link" id="' + serverResponse.date + '-reset">click' +
+		' here to send request now)</span>');
+
+	const testKeyPrefix = 'Test identifier: ';
+	const testKeyPostfix = ' get log via api/projects/{project}/sap-test-event/log/{sap-test-key}.';
+	text = text.replace(testKeyPrefix, '<span class="log-link" id="' + serverResponse.date + '-log" data-test-key="');
+	text = text.replace(testKeyPostfix, '">Click to retrieve/update log.</span>');
+
+	text.replace(':/CQSE/MSG_TIA:000 ', '');
+
+	let labelsClasses = 'labels';
+	if (serverResponse.status !== 200) {
+		labelsClasses += ' error';
+	}
+
+	return "<div class='" + labelsClasses + "'><div class='label'>" + serverResponse.action + "</div><div class='label'" +
+		" style='float:" +
+		" right'>" + serverResponse.date + "</div></div><p class='message'>" + text + "</p>";
+}
+
+function initializeLogLink(date) {
+	const logLink = document.getElementById(date + '-log');
+	if (!logLink) {
+		return;
+	}
+
+	const listener = {
+		elementId: logLink.id, listener: () => {
+			chrome.runtime.sendMessage({
+				data: {
+					request: 'get log',
+					sapTestKey: logLink.getAttribute('data-test-key')
+				}
+			});
+		}
+	};
+	listeners.push(listener);
+
+}
+
+function initializeResetLink(date) {
+	const resetLink = document.getElementById(date + '-reset');
+	if (!resetLink) {
+		return;
+	}
+
+	const listener = {
+		elementId: resetLink.id, listener: () => {
+			chrome.runtime.sendMessage({
+				data: {
+					request: 'reset user'
+				}
+			});
+		}
+	};
+	listeners.push(listener);
+
+}
+
+function makeObjectInspectionCollapsible(date) {
+	const infoElement = document.getElementById(date + '-info');
+	if (infoElement) {
+		listeners.push({elementId: infoElement.id, listener: () => triggerMoreInfo(date)});
+	}
+
+	const teaserElement = document.getElementById(date + '-teaser');
+	if (teaserElement) {
+		listeners.push({elementId: teaserElement.id, listener: () => triggerMoreInfo(date)});
+	}
```

Comment:
Would probably increase maintainability a bit because they seem to me like a potential code duplication candidate later on which we could maybe prevent in case this project gets bigger. 

But again just a small suggestion. You can also leave it as is :)